### PR TITLE
Add "Did You Mean This" Extension

### PR DESCRIPTION
**Pull request by christopher-o-toole**
Replacing the previous PR https://github.com/Azure/azure-cli-extensions/pull/1536

New extension (in preview) which uses the Aladdin service to generate and show failure recovery recommendations on command failure. This extension makes use of the hook added to `parser.py` in CLI version `2.4.0`. For more information, see [this CLI PR](https://github.com/Azure/azure-cli/pull/12889).

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [X] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S170

File: src/ai-did-you-mean-this/setup.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+setup(
+    name='ai_did_you_mean_this',
+    version=VERSION,
+    description='Recommend recovery options on failure.',
+    # TODO: Update author and email, if applicable
+    author="Christopher O'Toole",
+    author_email='chotool@microsoft.com',
+    # TODO: consider pointing directly to your source code instead of the generic repo
+    url='https://github.com/Azure/azure-cli-extensions',
```
Comment (haroldrandom) [inline src/ai-did-you-mean-this/setup.py:49]: Suggest to change this url to your folder where your README.md is placed directly which would give user a directl instruction to tell what this extension about.

File: src/ai-did-you-mean-this/setup.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# The full list of classifiers is available at
+# https://pypi.python.org/pypi?%3Aaction=list_classifiers
+CLASSIFIERS = [
+    'Development Status :: 4 - Beta',
+    'Intended Audience :: Developers',
+    'Intended Audience :: System Administrators',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.5',
```
Comment (haroldrandom) [inline src/ai-did-you-mean-this/setup.py:27]: Azure CLI Core doesn't support Python 3.5 anymore.
```suggestion
```

File: src/ai-did-you-mean-this/azext_ai_did_you_mean_this/custom.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            '-o': '--output',
+            '--only-show-errors': None,
+            '--help': None,
+            '--output': None,
+            '--query': None,
+            '--debug': None,
+            '--verbose': None
+        }
+
+        blacklisted = {'--debug', '--verbose'}
```
Comment (fengzhou-msft) [inline src/ai-did-you-mean-this/azext_ai_did_you_mean_this/custom.py:80]: Suggest to use `blocklist` just to avoid offensive language trouble.

Comment (jiasli) [reply src/ai-did-you-mean-this/azext_ai_did_you_mean_this/custom.py:80]: See https://bugs.chromium.org/p/chromium/issues/detail?id=981129

Comment (christopher-o-toole) [reply src/ai-did-you-mean-this/azext_ai_did_you_mean_this/custom.py]: Changed `blacklisted` to `blocklisted`.

**Pull request merged**